can-auth/vatican.c

```diff
@@ -211,10 +270,15 @@ int VULCAN_FUNC vulcan_send(ican_t *ican, uint16_t id, uint8_t *buf,
     /* 1. send legacy CAN message (ID | payload) */
     // NOTE: do not block for ACK, such that we can start the MAC computation
     rv = vatican_send(ican, id, buf, len, /*block=*/0);
-
+    
     /* 2. known authenticated connection ? send CAN authentication frame */
     if ((rv >= 0) && (vatican_mac_create(mac, id, buf, len) >= 0))
     {
+        #if VATITACAN
+            // Delay authentication message
+            sleep(encode_iat(vatican_cur->c));
```

Comment:
The `vatican_mac_create` function uses c+1 without committing it as a new nonce:

[https://github.com/sancus-pma/vulcan/blob/2169e4d4926ab4da40a030000817bdb861c380f3/can-auth/vatican.c#L75](url) 

so I think encoding c+1 here would cause the receiver to calculate the second MAC using c+2, which would be wrong? I might be missing something

Reply:
agreed it's a convention and it depends on what you encode over the channel. The MAC is created using c+1 and then it depends on the convention you use for the receiver. Your current code looks like the receiver expects the arrival time of a MAC with nonce n to encode (n-1), so that would be fine then (maybe add a comment).